superdesk/auth_server/clients.py

```diff
@@ -0,0 +1,182 @@
+# This file is part of Superdesk.
+#
+# Copyright 2019 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+import sys
+import logging
+from getpass import getpass
+import bcrypt
+import superdesk
+from superdesk.auth_server.scopes import allowed_scopes
+
+
+logger = logging.getLogger(__name__)
+
+
+class AuthServerClientsResource(superdesk.Resource):
+
+    schema = {
+        'client_id': {
+            'type': 'string',
+            'required': True,
+            'unique': True,
+        },
+        'password': {
+            'type': 'string',
+            'required': True
+        },
+        'scope': {
+            'type': 'list',
+            'allowed': list(allowed_scopes),
+            'required': True
+        }
+    }
+
+
+class AuthServerClientsService(superdesk.Service):
+    """Service to handle authorization server clients"""
+
+
+class RegisterClient(superdesk.Command):
+    """Register a client to authentication server
+
+    The client will be allowed to access the given scopes using the ``--scope`` argument
+    (this argument may be used several times).
+
+    A client id is needed, and a password will be prompted (without echo).
+
+    If the client already exists, it will be updated. You may use ``--password`` to update
+    the passsword, otherwise the former password is kept. Keep empty to prompt password,
+    this is better for security as the password won't appear on your screen, or in your
+    shell history or in your process list.
+
+    Examples:
+    Register a new client with id ``123456`` and allowed to reach items in archive and get users::
+
+        $ python manage.py auth_server:register_client 123456 -s ARCHIVE_READ -s USERS_READ
```

Comment:
imo would be better (for future UI) to register clients with only name and generate id and secret by superdesk

Reply:
hey. Well the `client_id` and `password` must be known by the client, so if SD generates them that means showing them, and that's not great to show the password in clear on the screen (that's the reason why I've implemented a prompt).

For future UI I was expecting an admin panel where SD client calls this API, in this case the ID and password can be generated by SD client it's not a problem.

On a side node there is currently no client name stored, but `client_id` is a free string, so we could use it to put a name.

Reply:
yep they must be known by the client, so that command must return it. but I don't know why the client would set a password, it's an app after all. here is how it's done by google, same is azure and onelogin at least, you set a name and get this:
![oauth](https://user-images.githubusercontent.com/179249/66988895-97f1b580-f0c3-11e9-8358-d29eee5fcfa3.png)

Reply:
so the password is shown in clear on the screen. I can do that, it's not a big change. Should I add a client name?

Reply:
it's shown clear on the screen. and the name you put in the form before this popup shows up.

Reply:
`password` and `client id` are now generated and `name` has been added.